classes/module/Module.php

```diff
@@ -3436,29 +3441,48 @@ public function isSymfonyContext()
     }
 
     /**
-     * Access the Symfony Container if we are in Symfony Context.
-     * Note: in this case, we must get a container from SymfonyContainer class.
-     * Note: if not in Symfony context, fallback to legacy Container for FO/BO.
+     * Access a service from the container (found from the controller or the context
+     * depending on cases). It uses ContainerFinder to find the appropriate container.
      *
      * @param string $serviceName
      *
-     * @return object|false if a container is not available, it returns false
+     * @return object|false If a container is not available it returns false
+     *
+     * @throws ServiceCircularReferenceException When a circular reference is detected
+     * @throws ServiceNotFoundException When the service is not defined
+     * @throws \Exception
      */
     public function get($serviceName)
     {
-        if ($this->isSymfonyContext()) {
-            if (null === $this->container) {
-                $this->container = SymfonyContainer::getInstance();
-            }
-
-            return $this->container->get($serviceName);
+        try {
+            $container = $this->getContainer();
+        } catch (ContainerNotFoundException $e) {
+            return false;
         }
 
-        if ($this->context->controller instanceof Controller) {
-            return $this->context->controller->get($serviceName);
+        return $container->get($serviceName);
+    }
+
+    /**
+     * Returns the container depending on the environment:
+     *  - Legacy: light container with few services specifically defined for legacy front/admin controllers
+     *  - Symfony: symfony container with all the migrated services (CQRS, ...)
+     *
+     * If you need to detect which kind of container you are using you can check if it is an instance of LegacyContainerInterface,
+     * which means it's a legacy/light container.
+     *
+     * @return LegacyContainerInterface|ContainerInterface
+     *
+     * @throws ContainerNotFoundException
+     */
+    public function getContainer()
```

Comment:
Why not set the return type?

```suggestion
    public function getContainer(): ContainerInterface
```

Reply:
Done